core/src/stages/compute_ik.cpp

```diff
@@ -352,9 +354,23 @@ void ComputeIK::compute() {
 		sandbox_scene->getCurrentState().copyJointGroupPositions(jmg, compare_pose);
 
 	double min_solution_distance = props.get<double>("min_solution_distance");
+	bool maximize_clearance = props.get<bool>("maximize_clearance");
+	// disable collision checking between links not in current group in order to compute
+	// clearance between links in current group and rest of the links
+	auto acm = sandbox_scene->getAllowedCollisionMatrix();
+	if (maximize_clearance) {
+		auto& all_link_names = sandbox_state.getRobotModel()->getLinkModelNames();
+		auto& group_link_names = jmg->getLinkModelNames();
+		std::vector<std::string> non_group_links;
+
+		for (const auto& link : all_link_names)
+			if (std::find(group_link_names.begin(), group_link_names.end(), link) == group_link_names.end())
+				non_group_links.push_back(link);
+		acm.setEntry(non_group_links, non_group_links, true);
```

Comment:
If you specify and `enableGroup` the active components in your distance request further down, there should be no need for this overhead at all. The collision interface ignores all checks which do not contain at least one active component.
Weren't you aware of this or was there some other reason why you went for the explicit matrix here?

Reply:
I wasn't aware of this. Is this documented somewhere?

Reply:
Well, it's ROS, so Read Our Sources. :angel: 

https://github.com/ros-planning/moveit/blob/master/moveit_core/collision_detection_fcl/src/collision_common.cpp#L426

The rest of the documentation I can find only refers to "only check active components", which can be interpreted in both ways.